src/code/z_camera.c

```diff
@@ -7602,19 +7607,22 @@ Vec3s Camera_Update(Camera* camera) {
         return camera->inputDir;
     }
 
-    // setting bgId to the ret of Quake_Calc, and checking that
-    // is required, it doesn't make too much sense though.
-    if ((bgId = Quake_Calc(camera, &quake), bgId != 0) && (camera->setting != CAM_SET_TURN_AROUND)) {
-        viewAt.x = camera->at.x + quake.atOffset.x;
-        viewAt.y = camera->at.y + quake.atOffset.y;
-        viewAt.z = camera->at.z + quake.atOffset.z;
-        viewEye.x = camera->eye.x + quake.eyeOffset.x;
-        viewEye.y = camera->eye.y + quake.eyeOffset.y;
-        viewEye.z = camera->eye.z + quake.eyeOffset.z;
+    // setting bgId to the ret of Quake_Update is required.
+    bgId = numQuakesApplied = Quake_Update(camera, &quakeCamData);
+
+    if ((numQuakesApplied != 0) && (camera->setting != CAM_SET_TURN_AROUND)) {
+        viewAt.x = camera->at.x + quakeCamData.atOffset.x;
+        viewAt.y = camera->at.y + quakeCamData.atOffset.y;
+        viewAt.z = camera->at.z + quakeCamData.atOffset.z;
+
+        viewEye.x = camera->eye.x + quakeCamData.eyeOffset.x;
+        viewEye.y = camera->eye.y + quakeCamData.eyeOffset.y;
+        viewEye.z = camera->eye.z + quakeCamData.eyeOffset.z;
+
         OLib_Vec3fDiffToVecSphGeo(&eyeAtAngle, &viewEye, &viewAt);
-        Camera_CalcUpFromPitchYawRoll(&viewUp, eyeAtAngle.pitch + quake.rotZ, eyeAtAngle.yaw + quake.unk_1A,
-                                      camera->roll);
-        viewFov = camera->fov + CAM_BINANG_TO_DEG(quake.zoom);
+        Camera_CalcUpFromPitchYawRoll(&viewUp, eyeAtAngle.pitch + quakeCamData.rollFromPitch,
+                                      eyeAtAngle.yaw + quakeCamData.rollFromYaw, camera->roll);
+        viewFov = camera->fov + CAM_BINANG_TO_DEG(quakeCamData.fov);
```

Comment:
since it already has "offset" in x/yOffset yeah it'd probably be better
otherwise I would think no "offset" in any arg could work, and you see it as "perturbation on x on y on fov on roll" ? maybe